kolibri/plugins/device/assets/src/views/DeviceSettingsPage/index.vue

```diff
@@ -236,6 +234,40 @@
             </div>
           </div>
         </div>
+
+        <div class="fieldset">
+          <h2>
+            {{ $tr('enabledPages') }}
+          </h2>
+          <p class="info-description">
+            {{ $tr('enabledPagesDescription') }}
+          </p>
+
+          <KCheckbox
```

Comment:
```suggestion
          <KCheckbox
            v-for="plugin in plugins"
            :label="plugin.name"
            :checked="plugin.enabled"
            :description="$tr('pluginDescription', { page: plugin.name })"
            @change="togglePlugin(plugin.id)"
          />
```
This would seem to suffice (except for not tracking that a restart could happen - that would have to dig into some other logic).

Reply:
I see two problems with this implementation (which is obviously nicer and I had tested it before submitting the deifinitive code):

- The way  `usePlugins` composable is built (which I don't like too much,  depends on a Promise to have the plugins, but the promise is not public for the composable user, so I had to add a watch to know when plugins had been read) would make the page blink when the plugins info are received, as there would not be any plugin when the component is mounted. This is the same reason not to use the translation provided by the api.
- It would automatically enable/disable the plugin, not giving the user the oportunity to cancel it if the user does not want to restart kolibri. Also, as you mentioned, the logic to avoid restarting kolibri three times if three plugins are changed would not work.

For the first point I think the change might be acceptable if you think the UX is not bad. For the `@change` event, I think it should be kept as it is now.

Reply:
> which I don't like too much, depends on a Promise to have the plugins, but the promise is not public for the composable user, so I had to add a watch to know when plugins had been read

Seems like the best thing to do here is to move the automatic fetching out of the composable, or to add a loading state ref in there, and use that to display a loading state for the plugins.

> For the @change event, I think it should be kept as it is now.

Yeah, I think doing additional tracking for the change event is fine, that was the part that I was most iffy about.

> It would automatically enable/disable the plugin

This could be reverted if they didn't want to, but yes, it would need additional tracking of state. Please do edit the composable for the needs here, this is the only place it is used, and I just put it in place to have a starting point for interfacing with the API.

Reply:
Changes done. Thanks for all the feedback and good ideas.